SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialDispatcher.cpp

```diff
@@ -201,3 +172,101 @@ Worker_ComponentId USpatialDispatcher::GetComponentId(Worker_Op* Op) const
 		return SpatialConstants::INVALID_COMPONENT_ID;
 	}
 }
+
+uint32 USpatialDispatcher::AddOpCallback(Worker_ComponentId ComponentId, const UserCallback<Worker_AddComponentOp>& Callback)
+{
+	return AddGenericOpCallback(ComponentId, WORKER_OP_TYPE_ADD_COMPONENT, [&](const Worker_Op* Op) {
+		Callback(Op->add_component);
+	});
+}
+
+uint32 USpatialDispatcher::AddOpCallback(Worker_ComponentId ComponentId, const UserCallback<Worker_RemoveComponentOp>& Callback)
+{
+	return AddGenericOpCallback(ComponentId, WORKER_OP_TYPE_REMOVE_COMPONENT, [&](const Worker_Op* Op) {
+		Callback(Op->remove_component);
+	});
+}
+
+uint32 USpatialDispatcher::AddOpCallback(Worker_ComponentId ComponentId, const UserCallback<Worker_AuthorityChangeOp>& Callback)
+{
+	return AddGenericOpCallback(ComponentId, WORKER_OP_TYPE_AUTHORITY_CHANGE, [&](const Worker_Op* Op) {
+		Callback(Op->authority_change);
+	});
+}
+uint32 USpatialDispatcher::AddOpCallback(Worker_ComponentId ComponentId, const UserCallback<Worker_ComponentUpdateOp>& Callback)
+{
+	return AddGenericOpCallback(ComponentId, WORKER_OP_TYPE_COMPONENT_UPDATE, [&](const Worker_Op* Op) {
+		Callback(Op->component_update);
+	});
+}
+
+uint32 USpatialDispatcher::AddOpCallback(Worker_ComponentId ComponentId, const UserCallback<Worker_CommandRequestOp>& Callback)
+{
+	return AddGenericOpCallback(ComponentId, WORKER_OP_TYPE_COMMAND_REQUEST, [&](const Worker_Op* Op) {
+		Callback(Op->command_request);
+	});
+}
+
+uint32 USpatialDispatcher::AddOpCallback(Worker_ComponentId ComponentId, const UserCallback<Worker_CommandResponseOp>& Callback)
+{
+	return AddGenericOpCallback(ComponentId, WORKER_OP_TYPE_COMMAND_RESPONSE, [&](const Worker_Op* Op) {
+		Callback(Op->command_response);
+	});
+}
+
+
+uint32 USpatialDispatcher::AddGenericOpCallback(Worker_ComponentId ComponentId, Worker_OpType OpType, TFunction<void(const Worker_Op*)> Callback) {
```

Comment:
Can the TFunction be const? not sure if there are any restrictions on passing TFunctions as const ref